### ISSUE TITLE
[Cloud_FireStore] Flutter plugin for firestore giving error, [Fatal Error] auto-value-annotations-1.6.5.pom:2:1: Content is not allowed in prolog.

### ISSUE THREAD

**Issue by afrozopsy** [FirebaseExtended/flutterfire#2973]
**Describe the bug**
The flutter plugin for cloud firestore v 0.14.0-dev.1 doesn't build on the given version of google-services:

com.google.gms:google-services:4.3.3

( This is the version of google-services that firebase tells to use in android apps. )

**To Reproduce**
1. First of all I'll attach the screenshots of my pubspec.yaml file, project level build.gradle file, and app level build.gradle files.

a> Pubspec.yaml file,

![1](https://user-images.githubusercontent.com/51536094/87777492-04f8ce00-c843-11ea-8744-fe31586daa53.PNG)

b> Project level build.gradle file,

![2](https://user-images.githubusercontent.com/51536094/87777532-180b9e00-c843-11ea-9f8c-fb60db723e10.PNG)

c> App level build.gradle file,

![3](https://user-images.githubusercontent.com/51536094/87777568-2b1e6e00-c843-11ea-8cbc-20cac9c1c43e.PNG)

**Expected behavior**
Plugin is supposed to work, but it throws an error during the build time.

**Additional context**
The error i get is as follows,

$ flutter run

```
Launching lib\main.dart on vivo 1906 in debug mode...
Running Gradle task 'assembleDebug'...
Plugin project :firebase_core_web not found. Please update settings.gradle.

Plugin project :cloud_firestore_web not found. Please update settings.gradle.

[Fatal Error] auto-value-annotations-1.6.5.pom:2:1: Content is not allowed in prolog.

[Fatal Error] auto-value-annotations-1.6.5.pom:2:1: Content is not allowed in prolog.

[Fatal Error] auto-value-annotations-1.6.5.pom:2:1: Content is not allowed in prolog.

[Fatal Error] auto-value-annotations-1.6.5.pom:2:1: Content is not allowed in prolog.



FAILURE: Build failed with an exception.



* What went wrong:

Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.

> Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.

   > Could not resolve com.google.auto.value:auto-value-annotations:1.6.5.

     Required by:

         project :app > com.google.firebase:firebase-common:19.3.0

      > Could not resolve com.google.auto.value:auto-value-annotations:1.6.5.

         > Could not parse POM https://jcenter.bintray.com/com/google/auto/value/auto-value-annotations/1.6.5/auto-value-annotations-1.6.5.pom

            > Content is not allowed in prolog.

      > Could not resolve com.google.auto.value:auto-value-annotations:1.6.5.

         > Could not parse POM https://jcenter.bintray.com/com/google/auto/value/auto-value-annotations/1.6.5/auto-value-annotations-1.6.5.pom

            > Content is not allowed in prolog.



* Try:

Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.



* Get more help at https://help.gradle.org



BUILD FAILED in 19s
Running Gradle task 'assembleDebug'...                             20.3s
Exception: Gradle task assembleDebug failed with exit code 1

```

**Flutter doctor**
Run `flutter doctor` and paste the output below:

Here is my flutter doctor -v output,

```
$ flutter doctor -v
[√] Flutter (Channel dev, 1.21.0-1.0.pre, on Microsoft Windows [Version 10.0.18363.900], locale en-PK)
    • Flutter version 1.21.0-1.0.pre at C:\Users\theVenom\00000_Flutter_Installed\flutter
    • Framework revision f25bd9c55c (2 days ago), 2020-07-14 20:26:01 -0400
    • Engine revision 99c2b3a245
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\theVenom\AppData\Local\Android\Sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\Users\theVenom\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[!] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.44.2)
    • VS Code at C:\Users\theVenom\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.9.1

[√] Connected device (1 available)
    • vivo 1906 (mobile) • bf93079e • android-arm64 • Android 9 (API 28)

! Doctor found issues in 2 categories.

```

**Comment by afrozopsy** [comment]
The issue seems to with coud_firestore plugin only, if i comment it in pubspec.yaml file and rebuild the project then it does run without any error. Also, the error is too vague, don't give off anything useful to work with. Please some maintainers help, been wrestling with it for 6 hours now.

**Comment by TahaTesser** [comment]
Hi @afrozopsy 
Looks like you're using `cloud_firestore:` which won't this version ` cloud_firestore: ^0.14.0-dev.1`. Since this isn't. a stable
Nevertheless, i just tried ` cloud_firestore: ^0.14.0-dev.1` with this and app runs fine
```
buildscript {
    ext.kotlin_version = '1.3.50'
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.5.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath 'com.google.gms:google-services:4.3.3'
    }
}

```

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel dev, 1.21.0-1.0.pre, on Mac OS X 10.15.6 19G73, locale en-GB)
    • Flutter version 1.21.0-1.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision f25bd9c55c (3 days ago), 2020-07-14 20:26:01 -0400
    • Engine revision 99c2b3a245
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (4 available)
    • SM M305F (mobile) • 32003c30dc19668f • android-arm64  • Android 10 (API 29)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 83.0.4103.116

• No issues found!

```

</details> 

Can you please a try new project and add your firebase config 
Thank you

**Comment by afrozopsy** [comment]
Thank you so much for replying. I wanted to mention that when i comment
this package in pubspec.yaml then the app runs fine. I did tried your
version here is the screenshot of my pubspec.yaml
[image: EmailThis.PNG]

Unfortunately, the error remains the same. Here is what error I am seeing.
I have reinstalled my flutter framework again. But to no luck.
[image: Fourth.PNG]
[image: Second.PNG]
[image: Third.PNG]

**Comment by afrozopsy** [comment]
Thank you so much for replying. I wanted to mention that when i comment this package in pubspec.yaml then the app runs fine. I did tried your version here is the screenshot of my pubspec.yaml

![EmailThis](https://user-images.githubusercontent.com/51536094/87801764-a7c44300-c869-11ea-8b34-39e865a29652.PNG)

Unfortunately, the error remains the same. Here is what error I am seeing. I have even reinstalled my flutter framework again. But to no luck.

![Fourth](https://user-images.githubusercontent.com/51536094/87801802-b4e13200-c869-11ea-9c30-7eedb20f4756.PNG)
![Second](https://user-images.githubusercontent.com/51536094/87801805-b6125f00-c869-11ea-8576-4ee819ed6f30.PNG)
![Third](https://user-images.githubusercontent.com/51536094/87801806-b7438c00-c869-11ea-9831-d879735ab17e.PNG)

**Comment by afrozopsy** [comment]
@TahaTesser can you see what this issue is actually coming from? I can't figure that out. Since I didn't wrote the plugin.

**Comment by TahaTesser** [comment]
Hi @afrozopsy 
Can you please provide `build.gradle` from android folder `build.gradle` from app folder inside android?
Thank you

**Comment by afrozopsy** [comment]
@TahaTesser thanks for help man! I talked to Michel Feinstein about this issue and he mentioned that it looks like it is related to my machine. And turns out he was right.

What had happened was that one of the cache file apparently auto-value-annotations-1.6.5.pom had corrupted on my machine and that corrupted file was causing the issue. And it was not getting re cached for some reason.

Solution:
I deleted my C:/UserName/.gradle folder. Everything in it. And the file was re cached and it fixed the error. Thanks everyone for help :) I'm going to close it now.
